### ISSUE TITLE
IE 11 - Method Map.prototype.set called on incompatible receiver 

### ISSUE THREAD

**Issue by roemhildtg** [canjs/canjs#4221]
I'm using the following polyfills, trying to get canjs to run in ie11. 

```
<script src="https://cdn.polyfill.io/v2/polyfill.min.js?features=Map,WeakMap,Set,WeakSet,Object.assign"></script>
```

But I'm running into the error message. 

![image](https://user-images.githubusercontent.com/5471079/41980017-14f9b39e-79eb-11e8-889d-944e56290313.png)

I think this is related: https://stackoverflow.com/questions/37199019/method-set-prototype-add-called-on-incompatible-receiver-undefined

**Comment by matthewp** [comment]
Can you back up the call stack a little bit and figure out what is attempted to be set?

**Comment by roemhildtg** [comment]
The value that's getting set is the can.getKeyValue symbol. Its calling the `assignSymbols` and iterating through.

This is what it looks like:

![image](https://user-images.githubusercontent.com/5471079/42039177-653fa526-7ab2-11e8-97a6-569e4cecd1f2.png)

**Comment by matthewp** [comment]
Any chance you go throw this in a glitch or jsbin or something?

**Comment by roemhildtg** [comment]
Sure - http://jsbin.com/pubiqoc/1/edit?html,js,output

It basically happens in _any_ polyfilled canjs application in IE11 I think.

**Comment by Mamuuut** [comment]
I had the same issue while upgrading CanJS from v3.x to v5.x.
We definitely need to support IE 11 for quite a lot of our customers unfortunately.
Is there any short term plan to make it happen?
Is there any way to help?

**Comment by justinbmeyer** [comment]
Yes, we are planning on making it work in about 2 weeks. Any help you can give would be great.

**Comment by Mamuuut** [comment]
Here are the problems I found:

1. in can-dom-mutate/can-dom-mutate (line 31):
```
	var data = dataStore.get(node) || dataStore.set(node, {}).get(node);
```
in IE 11 WeakMap.prototype.set is not chainable and returns undefined.

2. in can-reflect/types/map (line 24):
```
  shape.assignSymbols(Map.prototype, {
    "can.getOwnEnumerableKeys": Map.prototype.keys,
    "can.setKeyValue": Map.prototype.set,
    "can.getKeyValue": Map.prototype.get,
    "can.deleteKeyValue": Map.prototype["delete"],
    "can.hasOwnKey": Map.prototype.has
  });
```
this works for "can.getOwnEnumerableKeys" and then for "can.setKeyValue" but once "can.setKeyValue" symbol is assigned to Map.prototype.set, the error occurs.

In can-reflect/reflections/get-set/get-set (line 47 - getSetReflections.setKeyValue):
```
if(setKeyValue !== undefined) {
	return setKeyValue.call(obj, key, value);
} else {
	obj[key] = value;
}
```
setKeyValue is not undefined anymore and tries to call Map.prototype.set on Map.prototype which is not of type Map. This results in the Map polyfill to throw the error "Method Map.prototype.set called on incompatible receiver" (I tried 3 different ones, all throwing a similar error).

Not sure what the would be the best way to fix this issue.

Here is a glitch with an edited version of //unpkg.com/can@5/dist/global/core.js including my hack and a quick WeakMap.prototype.set polyfill to make it chainable.
https://flax-odometer.glitch.me/
